cli/flox-rust-sdk/src/providers/catalog.rs

```diff
@@ -123,23 +124,42 @@ pub struct CatalogClient {
 }
 
 impl CatalogClient {
-    pub fn new(baseurl: &str) -> Self {
+    pub fn new(baseurl: Option<&str>, device_uuid: Option<Uuid>, running_in_ci: bool) -> Self {
         // Remove the existing output file if it exists so we don't merge with
         // a previous `flox` invocation
         if let Ok(path_str) = std::env::var(FLOX_CATALOG_DUMP_DATA_VAR) {
             let path = Path::new(&path_str);
             let _ = std::fs::remove_file(path);
         }
 
+        use reqwest::header;
+        let mut headers = header::HeaderMap::new();
+        // Pass a flag if we are running in CI
+        if running_in_ci {
+            headers.insert("flox-ci", header::HeaderValue::from_static("true"));
+        };
+        // if given a device_uuid, send that along
+        if let Some(device_uuid) = device_uuid {
+            headers.insert(
+                "flox-device-uuid",
+                header::HeaderValue::from_str(&device_uuid.to_string())
+                    .expect("No device_id given!"),
+            );
+        }
+
         let client = {
             let timeout = std::time::Duration::from_secs(15);
             reqwest::ClientBuilder::new()
                 .connect_timeout(timeout)
                 .timeout(timeout)
                 .user_agent(format!("flox-cli/{}", &*FLOX_VERSION))
+                .default_headers(headers)
```

Comment:
Suggestion: I didn't know you could set default headers on the client itself. Since that's the case, just adding this before setting the default headers should also get you correlation for Sentry telemetry:
```rust
    if let Some(span) = sentry::configure_scope(|scope| scope.get_span()) {
        for (k, v) in span.iter_headers() {
            headers.insert(k, HeaderValue::from_str(&v).unwrap());
        }
    }
```